spec/models/services/facebook_spec.rb

```diff
@@ -22,11 +22,39 @@
       post_params = @service.create_post_params(post)
     end
 
-    it 'does not add post link when no photos' do
+    it 'adds URL of post to end if no link at end' do
       message = "Some text."
       post = double(message: double(plain_text_without_markdown: message, urls: []), photos: [])
       post_params = @service.create_post_params(post)
-      expect(post_params[:message]).not_to include "http"
+      expect(post_params[:message]).to include '(via http:'
+    end
+
+    it 'adds URL at the end with ending with valid HTTP url' do
+      message = "Some text and a link http://someserver1/testlink1 http://someserver2/testlink2 "
+      post = double(message: double(plain_text_without_markdown: message, urls: ["http://someserver1/testlink1", "http://someserver2/testlink2"]), photos: [])
+      post_params = @service.create_post_params(post)
+      expect(post_params[:message]).to include '(via http:'
+    end
+
+    it 'adds URL at the end with ending valid HTTPS url' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.